### PR TITLE
#289717 - fix RedirectAction and adding log on metadata value

### PR DESCRIPTION
**Pull request by mhnakashima**

### CODE REVIEW

**Comment by mhnakashima** [comment]
Added a redirect log validation

File: src/Take.Blip.Builder/Actions/Redirect/RedirectAction.cs [hunk 1]
```diff
@@ -2,26 +2,34 @@
...
         {
             _redirectManager = redirectManager;
+            _logger = logger;
         }
 
         public string Type => nameof(Redirect);
 
         public Task ExecuteAsync(IContext context, JObject settings, CancellationToken cancellationToken)
         {
+            var redirect = settings.ToObject<Lime.Messaging.Contents.Redirect>(LimeSerializerContainer.Serializer);
```
Comment (caiopsouza) [inline src/Take.Blip.Builder/Actions/Redirect/RedirectAction.cs:24]: Can you add a using for `Lime.Messaging.Contents` or is this disambiguation for another `Redirect`?

**Review by EricJangola — APPROVED** [review]

File: src/Take.Blip.Builder/Actions/Redirect/RedirectAction.cs [hunk 2]
```diff
@@ -2,26 +2,34 @@
...
+            context.Input.Message.Metadata.TryGetValue("REDIRECT_TEST_LOG", out var metadataRedirectAddress);
```
[the remark this answers is not in the document]
Comment (mhnakashima) [reply src/Take.Blip.Builder/Actions/Redirect/RedirectAction.cs]: fixed

Comment (andrebires) [inline src/Take.Blip.Builder/Actions/Redirect/RedirectAction.cs:28]: Metadata may be null here.

**Review by andrebires — COMMENTED** [review]
Please add at least a test to ensure that the solution is working

File: src/Take.Blip.Builder/Actions/Redirect/RedirectAction.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (rafael-santiago-take) [reply src/Take.Blip.Builder/Actions/Redirect/RedirectAction.cs]: ...

**Pull request merged**